### PR TITLE
Add configurable default guard

### PR DESCRIPTION
**Pull request by marijoo**
This Pull Request is backwards compatible and adds a new `default_guard` config option. 

This option is `null` by default. Setting it to `true` will always check permissions against the default guard from `auth.defaults.guard` if no explicit guard is passed. Users may also set this to a guard name which will then be used as the default.

This should address the issues from #1156.

### CODE REVIEW

**Comment by drbyte** [comment]
Note: The test suite is presently failing on Laravel 9 due to a bug in Laravel `9.36.0` and fixed in `9.36.2` which hasn't been tagged yet.

File: src/Traits/HasPermissions.php [hunk 1]
```diff
@@ -124,6 +124,8 @@ protected function convertToPermissionModels($permissions): array
      */
     public function filterPermission($permission, $guardName = null)
     {
+        $guardName = $guardName ?? $this->getDefaultGuardFromConfig();
```
[the remark this answers is not in the document]
Comment (marijoo) [reply src/Traits/HasPermissions.php]: @drbyte You are absolutely right. I removed the obsolete line. Thank you for pointing that out.

**Comment by parallels999** [comment]
Why did change code format style?
It makes this change hard to read

**Comment by marijoo** [comment]
> Why did change code format style?
> 
> It makes this change hard to read 

Argh, Prettier got in the way. Sorry, will fix this tommorow.

**Comment by marijoo** [comment]
@parallels999 I just removed the style changes. Sorry for the inconvenience.

**Pull request closed without merging**